src/utils/formatters.ts

```diff
@@ -190,6 +190,31 @@ module Plottable {
     }
 
     /**
+     * Transforms the Plottable TimeInterval into a d3 time interval equivalent
+     */
+    export function timeIntervalToD3Time(timeInterval: TimeInterval) {
+      switch (timeInterval) {
+      case TimeInterval.second:
+        return d3.time.second;
+      case TimeInterval.minute:
+        return d3.time.minute;
+      case TimeInterval.hour:
+        return d3.time.hour;
+      case TimeInterval.day:
+        return d3.time.day;
+      case TimeInterval.week:
+        return d3.time.week;
+      case TimeInterval.month:
+        return d3.time.month;
+      case TimeInterval.year:
+        return d3.time.year;
+      default:
+        Utils.Methods.warn("TimeInterval specified does not exist");
+        return d3.time.year;
```

Comment:
Probably nothing failed inside the try catch. If something fails (say this method has a bug, or the (JavaScript) user passes a wrong value into the axisConfiguration), then infinite loop (I believe). It happened when I mistyped some things in this method.